Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+
+                    //Power only starts affecting damage when it is above 5000
+                    damage = Math.Max(damage + (Math.Max(SMcomponent.Power - SupermatterComponent.PowerPenaltyThreshold, 0f)/500f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Molar count only starts affecting damage when it is above 1800
+                    damage = Math.Max(damage + (Math.Max(mixture.TotalMoles - SupermatterComponent.MolePenaltyThreshold, 0f)/80f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //There might be a way to integrate healing and hurting via heat
+			        //healing damage
+                    if (mixture.TotalMoles < SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        //Only has a net positive effect when the temp is below 313.15, heals up to 2 damage. Psycologists increase this temp min by up to 45
+                        var heatcap = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold) + (45 * SMcomponent.PsyCoeff));
+                        damage = Math.Max(-(2 * (1 - (mixture.Temperature / heatcap))),-2);
+                    }
+
+                    //if there are space tiles next to SM
+                    foreach(var adjacent in _atmosphereSystem.GetAdjacentTileMixtures(Xform.Coordinates))
+                    {
+                        if (adjacent.TotalMoles == 0)
+                        {
+                            if (integrity < 90)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 75)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0009f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 55)
+                                damage = Math.Clamp((SMcomponent.Power * 0.005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 25)
+                                damage = Math.Clamp((SMcomponent.Power * 0.002f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else
+                                break;
+                        }
+                    }
+                }
+
+                //only take up to 1.8 damage per cycle with no lower limmit
+                damage = Math.Min(SMcomponent.DamageArchived + (SupermatterComponent.DamageHardcap * SupermatterComponent.ExplosionPoint), damage);
+                //damage to add to total
+                var damageDelta = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Blunt"), (Shared.FixedPoint.FixedPoint2)damage);
```

Comment:
blunt is hardcoded?

Reply:
in tg this is all going into a ``damage`` var, so i just made the DamageSpecifier blunt for no particular reason